internal-packages/workflow-designer-ui/src/editor/properties-panel/text-generation-node-properties-panel/model/search-domain-filter-panel.tsx

```diff
@@ -145,53 +92,45 @@ export function SearchDomainFilterPanel({
 		const value = denylistInput.trim();
 		if (!value) return;
 		if (!DOMAIN_VALIDATION_REGEX.test(value)) return;
-		if (isMaxReached) return;
 		if (denylist.includes(`-${value}`) || allowlist.includes(value)) return;
+
+		console.log("Adding to denylist:", value);
 		updateDomainFilter(allowlist, [...denylist, `-${value}`]);
 		setDenylistInput("");
 	}
 
-	function removeAllowDomain(idx: number) {
-		const newAllow = allowlist.filter((_, i) => i !== idx);
-		updateDomainFilter(newAllow, denylist);
-	}
-
-	function removeDenyDomain(idx: number) {
-		const newDeny = denylist.filter((_, i) => i !== idx);
-		updateDomainFilter(allowlist, newDeny);
-	}
-
 	return (
-		<div className="search-domain-filter flex flex-col gap-2 mt-8">
-			<span className="filter-title font-bold text-[15px]">
+		<div className="search-domain-filter mt-8">
+			<div className="mb-4 text-[15px] font-medium text-white">
 				Search Domain Filter
-			</span>
+			</div>
 
-			<DomainList
-				type="allow"
-				domains={allowlist}
+			<SimpleDomainInput
+				label="Allow List"
 				inputValue={allowlistInput}
 				setInputValue={setAllowlistInput}
 				onAdd={addAllowDomain}
-				onRemove={removeAllowDomain}
-				isMaxReached={isMaxReached}
+				placeholder="Enter domain to include(e.g.,example.com)"
 			/>
 
-			<DomainList
-				type="deny"
-				domains={denylist}
+			{allowlist.length > 0 && (
+				<div className="ml-[150px] mb-4 text-[13px] text-gray-400">
+					Added: {allowlist.join(", ")}
+				</div>
+			)}
+
+			<SimpleDomainInput
+				label="Deny List"
 				inputValue={denylistInput}
 				setInputValue={setDenylistInput}
 				onAdd={addDenyDomain}
-				onRemove={removeDenyDomain}
-				isMaxReached={isMaxReached}
-				className="mt-4"
+				placeholder="Enter domain to exclude(e.g.,example.com)"
 			/>
 
-			{isMaxReached && (
-				<p className="max-domains-warning text-red-700 text-[12px]">
-					You can add up to {MAX_DOMAINS} domains only.
```

Comment:
I had limited the number of domains that could be specified to 10, but it seemed like the validation had disappeared.
I would be grateful if you could keep the process that checks the number of domains.

> [Limit Filter Size: You can add a maximum of 10 domains. Using fewer, more targeted domains often yields better results.](https://docs.giselles.ai/models/providers/perplexity#:~:text=Limit%20Filter%20Size%3A%20You%20can%20add%20a%20maximum%20of%2010%20domains.%20Using%20fewer%2C%20more%20targeted%20domains%20often%20yields%20better%20results.)